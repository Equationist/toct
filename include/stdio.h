/* stdio.h - Standard Input/Output Library for CompilerKit C Compiler */

#ifndef _STDIO_H
#define _STDIO_H

/* Standard I/O types */
typedef struct {
    int _fd;
    int _flags;
    char* _buffer;
} FILE;

/* Standard streams */
extern FILE* stdin;
extern FILE* stdout; 
extern FILE* stderr;

/* Standard I/O functions */
int printf(const char* format, ...);
int fprintf(FILE* stream, const char* format, ...);
int sprintf(char* str, const char* format, ...);
int snprintf(char* str, int size, const char* format, ...);

int scanf(const char* format, ...);
int fscanf(FILE* stream, const char* format, ...);
int sscanf(const char* str, const char* format, ...);

int putchar(int c);
int puts(const char* s);
int fputs(const char* s, FILE* stream);
int fputc(int c, FILE* stream);

int getchar(void);
char* gets(char* s);
char* fgets(char* s, int size, FILE* stream);
int fgetc(FILE* stream);
int ungetc(int c, FILE* stream);

FILE* fopen(const char* filename, const char* mode);
FILE* freopen(const char* filename, const char* mode, FILE* stream);
int fclose(FILE* stream);
int fflush(FILE* stream);

long ftell(FILE* stream);
int fseek(FILE* stream, long offset, int whence);
void rewind(FILE* stream);
int feof(FILE* stream);
int ferror(FILE* stream);
void clearerr(FILE* stream);

/* File position constants */
#define SEEK_SET 0
#define SEEK_CUR 1
#define SEEK_END 2

/* End of file constant */
#define EOF (-1)

/* Buffer size */
#define BUFSIZ 8192

/* Null pointer */
#ifndef NULL
#define NULL ((void*)0)
#endif

#endif /* _STDIO_H */