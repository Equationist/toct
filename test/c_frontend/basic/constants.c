/* Test various constant types */
#include <stdio.h>

int main() {
    // Integer constants
    int dec = 42;
    int hex = 0x2A;
    int oct = 052;
    
    // Character constants  
    char c = 'A';
    char newline = '\n';
    char tab = '\t';
    
    // String constants
    char* str = "Hello World";
    
    // Float constants
    float f = 3.14f;
    double d = 2.718;
    
    printf("Decimal: %d\n", dec);
    printf("Hex: %d\n", hex);
    printf("Octal: %d\n", oct);
    printf("Character: %c\n", c);
    printf("String: %s\n", str);
    printf("Float: %.2f\n", f);
    printf("Double: %.3f\n", d);
    
    return 0;
}