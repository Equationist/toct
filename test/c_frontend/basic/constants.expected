Decimal: 42
Hex: 42
Octal: 42
Character: A
String: Hello World
Float: 3.14
Double: 2.718